examples/example.hpp

```diff
@@ -1076,41 +1076,53 @@ bool device_with_streams(std::vector <rs2_stream> stream_requests, std::string&
 {
     rs2::context ctx;
     auto devs = ctx.query_devices();
-    for (auto type : stream_requests)
+    for (auto& dev : devs)
     {
-        bool stream_found = false;
-        for (auto& dev : devs)
+        std::map<rs2_stream, bool> found_streams;
+        for (auto& type : stream_requests)
         {
-            for (auto sensor : dev.query_sensors())
+            found_streams[type] = false;
+            for (auto& sensor : dev.query_sensors())
             {
-                for (auto profile : sensor.get_stream_profiles())
+                for (auto& profile : sensor.get_stream_profiles())
                 {
                     if (profile.stream_type() == type && dev.supports(RS2_CAMERA_INFO_SERIAL_NUMBER))
```

Comment:
Move the `&& dev.supports(RS2_CAMERA_INFO_SERIAL_NUMBER)` within the clause, as discussed